src/client.rs

```diff
@@ -775,6 +777,28 @@ impl Client {
         Ok(())
     }
 
+    /// Stream a single layer from an OCI registry.
+    ///
+    /// This is a streaming version of [`Client::pull_blob`].
+    /// Returns [`AsyncRead`](tokio::io::AsyncRead).
+    pub async fn async_pull_blob(
+        &self,
+        image: &Reference,
+        digest: &str,
+    ) -> Result<impl AsyncRead + Unpin> {
```

Comment:
I know this is already merged but @arronwy is there a reason you chose `AsyncRead` over returning `Stream`? Generally it is better to return a stream and let the user choose how they want to consume it

Reply:
Hi @thomastaylor312 ， yes, agree, `Stream` can easy convert to `AsyncRead` I also verified if we return 
`Result<impl Stream<Item = std::result::Result<bytes::Bytes, std::io::Error>>>`
will also works. 

The reason I chose `AsyncRead` is try to consistent with current `pull_blob` API with output using:
`AsyncWrite` and `async_compression` crates: https://docs.rs/async-compression/latest/async_compression/index.html also operate over `AsyncWrite` and `AsyncBufRead` streams.

It's ok for me if we decide change to return `Stream` to let user to choose :-).

Reply:
Right I had forgotten we used `AsyncRead`. I'll capture an issue to refactor that at some point